### ISSUE TITLE
RFE: work with backlinks

### ISSUE THREAD

**Issue by juray** [glensc/dokuwiki-plugin-pageredirect#3]
Suppose, there are synonymic pagenames and other wiki pages refer to all (with roughly equal counts). With redirect all such links lead to one "target" page, well.

What is wanted: view on backlinks search all pages refer to all  this synonyms (now links to this page found only directly refer to this target page)

Is this possible by modify this plugin or need to rewrite dokuwiki core?

**Comment by glensc** [comment]
@juray can you make example which page should be related to which page. i.e describe three pages, their content and their inner relations. say page a, page b, page c.

**Comment by glensc** [comment]
i suppose for backlinks need to set `$metadata[relation][references]` list of page ids?

**Comment by michitux** [comment]
@glensc: yes, that is the information that is used by the indexer. I think a possible scenario is the following: page A links to page B. However, page B redirects to C. When viewing backlinks of page C, A should be shown. As I said in my previous comment, this could be done by modifying the information of page A when it is added to the index.

**Comment by glensc** [comment]
@michitux do you have example of project which unit tests this?

also the relation.references should be updated not replaced. how to ensure it doesn't get stale if i just "add" items there. because i can't remove items from the list, they could be added there by other plugins.
